### PR TITLE
Added documentation

### PR DESCRIPTION
**Pull request by alexsporn**

### CODE REVIEW

**Review by muXxer — APPROVED** [review]

File: documentation/docs/welcome.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+## Setup
+
+The recommended setup is to use the provided [Docker images](https://hub.docker.com/r/iotaledger/inx-indexer).
+These images are also used in our [HORNET recommended setup using Docker](http://wiki.iota.org/hornet/develop/how_tos/using_docker).
+
+## Configuration
+
+The indexer is configured by default to connect to your HORNET instance.<br />
+You can find all the configuration options [here](configuration.md).
```
Comment (lucas-tortora) [inline documentation/docs/welcome.md]:
```suggestion
The indexer is configured by default to connect to your HORNET instance.

You can find all the configuration options in the [configuration section](configuration.md).
```

File: documentation/docs/welcome.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+## API
+
+The indexer exposes a custom set of REST APIs that can be used by wallets and applications to find UTXO in the ledger with a given query.<br />
+You can find more information about the API [here](api_reference.md).
```
Comment (lucas-tortora) [inline documentation/docs/welcome.md]:
```suggestion
The indexer exposes a custom set of REST APIs that can be used by wallets and applications to find UTXO in the ledger with a given query.

You can find more information about the API in the [API reference section](api_reference.md).
```

File: .github/workflows/test-docs-build.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+jobs:
+  checks:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Test Build
+        run: |
+          cd documentation
+          yarn install --immutable
+          yarn build
```
Comment (Dr-Electron) [inline .github/workflows/test-docs-build.yml]:
```suggestion
      - uses: actions/setup-node@v3
        with:
          node-version: 16
          cache: 'yarn'
      - name: Allow modern Yarn
        run: |
          corepack enable
      - name: Test Build
        working-directory: documentation
        run: |
          yarn install --immutable
          yarn build
```

**Review by lucas-tortora — APPROVED** [review]
Just some really minor comments.  LGTM! :)

File: documentation/.yarnrc.yml [hunk 4]
```diff
@@ -0,0 +1,3 @@
+nodeLinker: node-modules
+
+yarnPath: .yarn/releases/yarn-3.2.0.cjs
```
Comment (Dr-Electron) [inline documentation/.yarnrc.yml]:
```suggestion
nodeLinker: node-modules

```

You can remove the `.yarn` folder now as we use the new `core pack enable` to make sure the correct version of yarn is used

File: documentation/README.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
+# Documentation
+
+The documentation is built using [Docusaurus 2](https://docusaurus.io/). The deployment is done through a centralized build from [IOTA WIKI](https://github.com/iota-community/iota-wiki). To run a local instance the [IOTA WIKI CLI](https://github.com/iota-community/iota-wiki-cli) is used.
+
+## Prerequisites
+
+- [Node.js v14.14+](https://nodejs.org/en/)
```
Comment (Dr-Electron) [inline documentation/README.md]:
```suggestion
- [Node.js v16.10+](https://nodejs.org/en/) and run `corepack enable` to enable modern Yarn.
```

File: documentation/sidebars.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+                },
+                {
+                    type: 'doc',
+                    id: 'api_reference',
+                    label: 'API Reference',
+                }
+            ]
+        }
+    ]
+};
```
Comment (Dr-Electron) [inline documentation/sidebars.js]:
```suggestion
};

```

:trollface:

**Review by Dr-Electron — CHANGES_REQUESTED** [review]
LGTM. Just change to `corepack enable` so that we don't need to create another PR for that and we should be fine.

File: .github/workflows/test-docs-build.yml [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+jobs:
+  checks:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 16
+          cache: 'yarn'
```
Comment (Dr-Electron) [inline .github/workflows/test-docs-build.yml:20]:
```suggestion
          cache: 'yarn'
          cache-dependency-path: documentation/yarn.lock
```

**Pull request merged**